woodwork/column_accessor.py

```diff
@@ -192,15 +197,14 @@ def wrapper(*args, **kwargs):
 
                 # Try to initialize Woodwork with the existing Schema
                 if _is_series(result):
-                    valid_dtype = _get_valid_dtype(result, self._schema['logical_type'])
-                    if result.dtype == valid_dtype:
+                    valid_dtype = _get_valid_underlying_dtype(result, self._schema['logical_type'])
+                    if str(result.dtype) == valid_dtype:
                         schema = copy.deepcopy(self._schema)
-                        # We don't need to pass dtype from the schema to init
-                        del schema['dtype']
                         result.ww.init(**schema, use_standard_tags=self.use_standard_tags)
                     else:
+                        user_dtype = _get_valid_ltype_dtype(result, self._schema['logical_type'])
```

Comment:
felt this case was similar to how we tell users to convert to the `convert_dtype` dtype instead of the `valid_dtype` dtype, but I know it's a slightly different situation here, so happy to switch back.

Reply:
Now that I think about it more, I agree that it makes the most sense that we show the underlying dtype here. It might make it confusing if a user wants to do the type conversion themself, but if they're using koalas, they likely are familiar enough with koalas to know that they'll need to convert with `str`